src/main/java/org/opentripplanner/routing/api/request/preference/SystemPreferences.java

```diff
@@ -2,35 +2,54 @@
 
 import java.io.Serializable;
 import java.time.Duration;
-import java.util.HashSet;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+import java.util.Objects;
 import java.util.Set;
-import javax.annotation.Nonnull;
+import java.util.function.Consumer;
 import org.opentripplanner.ext.dataoverlay.api.DataOverlayParameters;
-import org.opentripplanner.routing.api.request.ItineraryFilterParameters;
 import org.opentripplanner.routing.api.request.RoutingTag;
+import org.opentripplanner.util.lang.ToStringBuilder;
+
+/**
+ * Configure system related features - a system feature is an none-functional feature. It
+ * describes how the system should work, but not change the output of a travel request.
+ * <p>
+ * Some parameters in this class are related to functional-features, but does not have a clear
+ * place where they belong. We should refactor and move these.
+ * <p>
+ * THIS CLASS IS IMMUTABLE AND THREAD-SAFE
+ */
+public class SystemPreferences implements Serializable {
+
+  public static final SystemPreferences DEFAULT = new SystemPreferences();
+
+  private final Set<RoutingTag> tags;
```

Comment:
`tags` should be moved to request, probably also `dataOverlay`. We can do this at a later stage.

Reply:
Agree, the `tags` should go in the request, a bit unsure about the data-overlay, there might be a mix of preferences and request parameters in it.